assets/js/components/FilterPanel.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+module.exports = class FilterPanel {
+
+  constructor($elm, _window = window, doc = document) {
+    if (!$elm) {
+      return;
+    }
+
+    this.$elm = $elm;
+    this.window = _window;
+    this.doc = doc;
+
+    this.$button = $elm.querySelector('button');
+
+    if (!this.$button.form) {
```

Comment:
The pattern doesn't define a form, so make sure it exists before doing anything.